### ISSUE TITLE
Testing plans

### ISSUE THREAD

**Issue by koustuvsinha** [zippyui/react-datagrid#11]
Hi! I would like to know what are the current testing plans as given in the roadmap. Would like to learn and contribute :)

**Comment by zippyui-react** [comment]
We plan integrating CI, as we've done with https://github.com/zippyui/react-date-picker.

See https://github.com/zippyui/react-date-picker/blob/master/circle.yml and https://github.com/zippyui/react-date-picker/tree/master/test

Circleci for date-picker is found at https://circleci.com/gh/zippyui/react-date-picker, notice master branch being successful

**Comment by zippyui-react** [comment]
So basically we want to integrate with mocha/should/circleci for continuous integration. Then take every public property of the date picker and make a test to ensure it behaves/renders as expected

**Comment by koustuvsinha** [comment]
great! will start looking into it right away. please advise which properties to start with, and the ones you are currently working on

**Comment by zippyui-react** [comment]
We are currently working on remote pagination.

If you are willing to help, you could add a basic test directory, add mocha and should as dev deps in package.json and have a simple test running as a proof. Then we can expand from here. I'll try and make a test plan and share it with you today or tomorrow. After this, we can integrate this with CI, and follow the test plan.
Thanks for contributing!

**Comment by koustuvsinha** [comment]
Sure! I had a question though (forgive me if it sounds too noob). Why are you using Makefile to run test instead of, say Karma test runner? The datepicker tests are not working in windows environment (will try it again shortly in ubuntu) and giving the following error : 

`./node_modules/.bin/mocha --recursive --reporter spec --require should
'.' is not recognized as an internal or external command,
operable program or batch file.
make: *** [test] Error 1
npm ERR! Test failed.  See above for more details.`

**Comment by koustuvsinha** [comment]
Ok it was due to the directory writing style of windows and unix, unfortunately windows cannot understand the backslashes! It is a trivial issue, still I would like to know whether any platform independent solution exists or not :)

**Comment by koustuvsinha** [comment]
Also, why didn't you just put the test script `./node_modules/.bin/mocha --recursive --reporter spec --require should` directly into npm test, because of iojs?

**Comment by zippyui-react** [comment]
We've added a basic test plan at https://github.com/zippyui/react-datagrid/blob/master/TEST-PLAN.md

We're going to integrate CI Friday, 24 april 2015.
We appreciate your effort! Currently we are busy working on remote data support for the grid and pagination, as we plan releasing them next week

**Comment by zippyui-react** [comment]
We've just integrated CI, have a look at https://circleci.com/gh/zippyui/react-datagrid

**Comment by koustuvsinha** [comment]
Ok I am working on the test cases. May I break the test cases into separate files logically?

**Comment by zippyui-react** [comment]
Yes, absolutely, that would be preferable :) Different files even different folders (for which you may need to specify mocha to traverse the test folder recorsively)

**Comment by koustuvsinha** [comment]
Having a bit difficulty in simulating the column menu click event, when I try to isolate the element by this way : 
```
        var columnMenuBtnArray = tryWithClass(table,'z-show-menu');
        columnMenuBtnArray.should.not.be.empty;    // ok upto here
        
        var columnMenuBtn = columnMenuBtnArray[0].getDOMNode();  
        //trying to simulate the click
        TestUtils.Simulate.mouseDown(columnMenuBtn);    // tried with click earlier
        TestUtils.Simulate.mouseUp(columnMenuBtn);

        var colSelectedRows = tryWithClass(table,'menu-row');
        // selectedRows turns out to be a blank array, which shouldn't be the case

```
whichever way I try, the click is not creating the menu-row classes for the menu, neither is `z-menu` or `z-menu-scroll-container` class is created. A little help would be great :)

**Comment by koustuvsinha** [comment]
on the contrary, I can easily simulate the sorting event by the following : 

```
        var headers = tryWithClass(table,'z-column-header'');
        var header = headers[0].getDOMNode();
        TestUtils.Simulate.mouseDown(header);
        TestUtils.Simulate.mouseUp(header);
```
I might be missing some special class for the column menu display

**Comment by zippyui-react** [comment]
I've just added a basic test for multiple selection.

Please review the commit at https://github.com/zippyui/react-datagrid/commit/9c83d07a1fb8f82df04b3de9f910536ec7117c3f as I fix some code-style issues.

I've not setup a js code style standard yet, but for now, please leave an empty space before values:

```jsx
var a=4
function name(a,b){
}
```
becomes

```jsx
var a = 4
function name(a, b){
}
```
Also, please avoid semicolons. Only use them before [] and (). If/else always need curly braces.

**Comment by koustuvsinha** [comment]
sure! you might want to look at [feross/standard](https://github.com/feross/standard) if you want to set up code style standard
